python/lsst/afw/cameraGeom/basicCameraFactory.py

```diff
@@ -0,0 +1,233 @@
+            for l in fh:
+                if l.startswith("#"):
+                    continue
+                detConfig = DetectorConfig()
+                els = l.rstrip().split()
+                detConfig.name = els[0]
+                detConfig.id = self._detectorIdFromAbbrevName(els[0])
+                detConfig.bbox_x0 = 0
+                detConfig.bbox_y0 = 0
+                detConfig.bbox_x1 = int(els[5]) - 1
+                detConfig.bbox_y1 = int(els[4]) - 1
+                detConfig.detectorType = self._detTypeMap[els[6]]
+                detConfig.serial = els[0]
+
+                # Convert from microns to mm.
+                detConfig.offset_x = float(els[1])/1000.
+                detConfig.offset_y = float(els[2])/1000.
+
+                detConfig.refpos_x = (int(els[5]) - 1.)/2.
+                detConfig.refpos_y = (int(els[4]) - 1.)/2.
+                # TODO translate between John's angles and Orientation angles.
+                # It's not an issue now because there is no rotation except about z in John's model.
+                detConfig.yawDeg = 90.*nQuarter + float(els[7])
+                detConfig.pitchDeg = float(els[8])
+                detConfig.rollDeg = float(els[9])
+                detConfig.pixelSize_x = float(els[3])/1000.
+                detConfig.pixelSize_y = float(els[3])/1000.
+                detConfig.transposeDetector = False
+                detConfig.transformDict.nativeSys = PIXELS.getSysName()
+                # The FOCAL_PLANE and TAN_PIXEL transforms are generated by the Camera maker,
+                # based on orientaiton and other data.
+                # Any additional transforms (such as ACTUAL_PIXELS) should be inserted here.
+                detectorConfigs.append(detConfig)
+        return detectorConfigs
+
+
+    def _makeCameraData(self):
+        """
+        Create the underlying data used to generate the camera
+        """
+
+        detectorConfigList = self.makeDetectorConfigs()
+        schema = afwTable.AmpInfoTable.makeMinimalSchema()
+        ampTableDict = {}
+        for detConfig in detectorConfigList:
+            ampCatalog = afwTable.AmpInfoCatalog(schema)
+            ampTableDict[detConfig.name] = ampCatalog
+
+        #Build the camera config.
+        camConfig = CameraConfig()
+        camConfig.detectorList = dict([(i,detectorConfigList[i]) for i in xrange(len(detectorConfigList))])
+        camConfig.name = self._cameraName
+        camConfig.plateScale = 1.0/afwGeom.radToArcsec(self._radialTransform[1])
+
+        tConfig = afwGeom.TransformConfig()
+        tConfig.transform.name = 'inverted'
+        radialClass = afwGeom.xyTransformRegistry['radial']
+        tConfig.transform.active.transform.retarget(radialClass)
+        tConfig.transform.active.transform.coeffs = self._radialTransform
+        tmc = afwGeom.TransformMapConfig()
+        tmc.nativeSys = FOCAL_PLANE.getSysName()
+        tmc.transforms = {PUPIL.getSysName():tConfig}
```

Comment:
This looks a bit cramped. Please add a space after the colon (which is also the PEP8 style).